appserver/jdbc/jdbc-runtime/src/main/java/org/glassfish/jdbcruntime/deployment/annotation/handlers/DataSourceDefinitionHandler.java

```diff
@@ -394,11 +394,12 @@ private DataSourceDefinitionDescriptor createDescriptor(DataSourceDefinition def
             desc.setDatabaseName(defn.databaseName());
         }
 
-        if ((desc.getPortNumber() != -1 && desc.getDatabaseName() != null && desc.getServerName() != null)) {
+        if (desc.getPortNumber() != -1 && desc.getDatabaseName() != null && desc.getServerName() != null) {
             //standard properties are set, ignore URL
         } else {
             if (defn.url() != null && !defn.url().equals("")) {
                 desc.setUrl(defn.url());
+                desc.setServerName(null); //To prevent serverName overriding the URL, always use the URL if the standard properties are not set
```

Comment:
Can you make a small change to the `DataSourceDefinitionDescriptor.setServerName(String serverName)` method please to reflect this.

That method sets a boolean of `serverNameSet` to true, which when you're setting it to null probably shouldn't be the case :)